From 5a4aba7b67b6d2bd75cca746c576b7fc73765199 Mon Sep 17 00:00:00 2001
From: MengLi <meng.li@windriver.com>
Date: Mon, 30 Mar 2020 14:27:10 +0800
Subject: [PATCH 02/22] driver: media: imx8: replace dma_zalloc_coherent with
 dma_alloc_coherent

According to upstream commit  dfd32cad146e
("dma-mapping: remove dma_zalloc_coherent()"),
dma_zalloc_coherent had been removed, and the function
dma_alloc_coherent already has the operation of clearing
memory.

Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/platform/imx8/mxc-jpeg.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/media/platform/imx8/mxc-jpeg.c b/drivers/media/platform/imx8/mxc-jpeg.c
index c851b66aedd2..85a5a1e8e395 100644
--- a/drivers/media/platform/imx8/mxc-jpeg.c
+++ b/drivers/media/platform/imx8/mxc-jpeg.c
@@ -437,21 +437,21 @@ static bool mxc_jpeg_alloc_slot_data(struct mxc_jpeg_dev *jpeg,
 	if (jpeg->slot_data[slot].desc)
 		goto skip_alloc; /* already allocated, reuse it */
 	/* allocate descriptor for decoding/encoding phase */
-	jpeg->slot_data[slot].desc = dma_zalloc_coherent(jpeg->dev,
+	jpeg->slot_data[slot].desc = dma_alloc_coherent(jpeg->dev,
 		sizeof(struct mxc_jpeg_desc),
 		&jpeg->slot_data[slot].desc_handle, GFP_ATOMIC);
 	if (!jpeg->slot_data[slot].desc)
 		goto err;
 
 	/* allocate descriptor for configuration phase (encoder only) */
-	jpeg->slot_data[slot].cfg_desc = dma_zalloc_coherent(jpeg->dev,
+	jpeg->slot_data[slot].cfg_desc = dma_alloc_coherent(jpeg->dev,
 		sizeof(struct mxc_jpeg_desc),
 		&jpeg->slot_data[slot].cfg_desc_handle, GFP_ATOMIC);
 	if (!jpeg->slot_data[slot].cfg_desc)
 		goto err;
 
 	/* allocate configuration stream */
-	jpeg->slot_data[slot].cfg_stream_vaddr = dma_zalloc_coherent(
+	jpeg->slot_data[slot].cfg_stream_vaddr = dma_alloc_coherent(
 		jpeg->dev,
 		MXC_JPEG_MAX_CFG_STREAM,
 		&jpeg->slot_data[slot].cfg_stream_handle, GFP_ATOMIC);
-- 
2.17.1

